lib/sqewer/extensions/appsignal_wrapper.rb

```diff
@@ -35,7 +35,9 @@ def around_deserialization(serializer, msg_id, msg_payload)
 
         transaction.set_action('%s#%s' % [serializer.class, 'unserialize'])
         transaction.request.params = {:sqs_message_body => msg_payload.to_s}
-        transaction.set_http_or_background_queue_start
+        if msg_attributes.key?('SentTimestamp')
+          transaction.set_queue_start = Time.at(msg_attributes['SentTimestamp'].to_i / 1000)
```

Comment:
You might want to use a float division here because you are chopping away the subsecond precision which the timestamp does offer.

Reply:
I just copied [this](https://github.com/appsignal/appsignal-ruby/blob/master/lib/appsignal/hooks/shoryuken.rb#L22). It's not clear from Appsignal's docs what does this arg support.

Do you want to try it in millis? Then we can just remove the division.

Reply:
It accepts a Time object, and the Time object has subsecond precision available. So all you need to do is divide by `1000.to_f` instead of just `1000`

Reply:
Cool cool!